src/Controllers/lessonsPlans.controller.ts

```diff
@@ -3,12 +3,56 @@ import { LessonsPlanEntry } from '../Types/lessonsPlanEntry.type';
 import { ErrorType } from '../Types/error.type';
 import { LessonsModel } from '../Models/lessonPlanEntry.model';
 
-export const getAllLessonsPlans: RequestHandler = async (
+export const getAllLessonsForMajorYear: RequestHandler = async (
     req: Request,
     res: Response<LessonsPlanEntry[] | ErrorType>
 ) => {
     try {
-        const lessonsPlans = await LessonsModel.find({}, { __v: 0 });
+        const { major, year } = req.params;
+
+        const lessonsPlans = await LessonsModel.find(
+            { name: major, year: year },
+            { __v: 0 }
+        );
+
+        return res.status(200).json(lessonsPlans);
+    } catch (error) {
+        return res.status(500).json({
+            status: 500,
+            message:
+                'Something went wrong when retrieving lessons from database',
+        });
+    }
+};
+
+export const getAllLessonsForMajorYearGroup: RequestHandler = async (
+    req: Request,
+    res: Response<LessonsPlanEntry[] | ErrorType>
+) => {
+    try {
+        const { major, year, group } = req.params;
+
+        if (group === 'lectures') {
+            const lessonsPlans = await LessonsModel.find(
+                {
+                    name: major,
+                    year: year,
+                    groups: 'all',
+                },
+                { __v: 0 }
+            );
+
+            return res.status(200).json(lessonsPlans);
```

Comment:
I think that it would be better to create new controller to handle requests only for lectures

Reply:
resolved